client/shared/src/search/query/decoratedToken.ts

```diff
@@ -81,21 +81,34 @@ export interface MetaField extends BaseMetaToken {
 }
 
 /**
- * A token that is labeled and interpreted as repository revision syntax.
+ * A token that is labeled and interpreted as repository revision syntax in Sourcegraph. Note: there
+ * are syntactic differences from pure Git ref syntax.
  * See https://docs.sourcegraph.com/code_search/reference/queries#repository-revisions.
  */
 export interface MetaRevision extends BaseMetaToken {
     type: 'metaRevision'
     kind: MetaRevisionKind
 }
 
-export enum MetaRevisionKind {
-    Separator = 'Separator', // is a ':'
-    Label = 'Label', // a branch or tag
+export type MetaRevisionKind = MetaGit | MetaSourcegraph
+
+/**
+ * A custom revision syntax that is only valid in Sourcegraph search queries.
+ */
+export enum MetaSourcegraph {
+    Separator = 'Separator', // a ':' that separates revision patterns
+    IncludeGlobMarker = 'IncludeGlobMarker', // a '*' at the beginning of a revision pattern to mark it as an "include" glob pattern.
+    ExcludeGlobMarker = 'ExcludeGlobMarker', // a '*!' at the beginning of a revision pattern to mark it as an "exclude" glob pattern.
+}
```

Comment:
```suggestion
export enum MetaSourcegraphRevision {
    Separator = 'Separator', // a ':' that separates revision patterns
    IncludeGlobMarker = 'IncludeGlobMarker', // a '*' at the beginning of a revision pattern to mark it as an "include" glob pattern.
    ExcludeGlobMarker = 'ExcludeGlobMarker', // a '*!' at the beginning of a revision pattern to mark it as an "exclude" glob pattern.
}
```

Nit: a more specific name would make this explicit that this enum solely refers to revision-specific syntax (and not **all** Sourcegraph syntax)

Reply:
cool, added it to get `MetaGitRevision` too.